rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/KubernetesApiImpl.scala

```diff
@@ -311,4 +353,122 @@ PUTs must contain resourceVersions. Response:
     } yield resource
   }
 
+  private val pathForGetPod
+      : Uri.Path = Uri.Path.Empty / "api" / "v1" / "namespaces" / namespace / "pods" / settings.podName
+
+  private def pathForReplicaSet(replicaSetName: String): Uri.Path =
+    Uri.Path.Empty / "apis" / "apps" / "v1" / "namespaces" / namespace / "replicasets" / replicaSetName
+
+  private def getReplicaSet(name: String): Future[ReplicaSet] = {
+    val httpResponse = makeRequest(requestForPath(pathForReplicaSet(name)), s"Timeout getting replica set '$name'")
+    for {
+      response <- httpResponse
+      responseEntity <- response.entity.toStrict(settings.bodyReadTimeout)
+      replicaSet <- response.status match {
+        case StatusCodes.OK =>
+          Unmarshal(responseEntity).to[ReplicaSet].recover {
+            case ex =>
+              throw new PodListException(s"Error while parsing ReplicaSet: ${ex.getMessage}")
+          }
+        case StatusCodes.Unauthorized =>
+          handleUnauthorized(response)
+        case unexpected =>
+          responseEntity
+            .toStrict(settings.bodyReadTimeout)
+            .flatMap(e => Unmarshal(e).to[String])
+            .map(body =>
+              throw new ReplicaSetException(
+                s"Unexpected response from API server when creating PodCost StatusCode: $unexpected. Body: $body"))
+      }
+    } yield {
+      replicaSet
+    }
+  }
+
+  private def getPod(): Future[Pod] = {
+    val httpResponse = makeRequest(requestForPath(pathForGetPod), "Timeout listing pods")
+    for {
+      response <- httpResponse
+      responseEntity <- response.entity.toStrict(settings.bodyReadTimeout)
+      podList <- response.status match {
+        case StatusCodes.OK =>
+          Unmarshal(responseEntity).to[Pod].recover {
+            case ex => throw new PodListException(s"Error while parsing PodList: ${ex.getMessage}")
+          }
+        case StatusCodes.Unauthorized =>
+          handleUnauthorized(response)
+        case unexpected =>
+          Unmarshal(response.entity)
+            .to[String]
+            .map(body =>
+              throw new PodListException(
+                s"Unexpected response from API server when retrieving PodCost StatusCode: $unexpected. Body: $body"))
+      }
+    } yield podList
+  }
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object KubernetesApiImpl {
```

Comment:
If you are ok with this refactoring, I will change the other instantiation of `KubernetesApiImpl` in PostDeletionCost as well.

Reply:
👍